tools/src/tester/helpers.ts

```diff
@@ -0,0 +1,8 @@
+import { type Evaluation, Result } from './types/eval.types'
+
+export function overall_result (evaluations: Evaluation[]): Result {
+  if (evaluations.some(e => e.result === Result.ERROR)) return Result.ERROR
+  if (evaluations.some(e => e.result === Result.FAILED)) return Result.FAILED
+  if (evaluations.some(e => e.result === Result.SKIPPED)) return Result.SKIPPED
```

Comment:
This is for evaluating components results. A component is skipped when a previous component results in an error (say if a chapter could not be executed for validation, all subsequent chapters will be skipped). However that would result in `ERROR` for overall. So for now line 6 is actually redundant. 

Hypothetically when we allow the user to skip some component evaluation (say skip evaluating the response body of a new endpoint still being developed), I think we should still mark the parent component as skipped. The print out will show all subcomponents so you can tell which subcomponent causes the `SKIPPED` status for the parent component. So:
- If you see ERROR as the result for a component, at least one of its subcomponent has ERROR
- otherwise, If you see FAILED as  the result for a component, at least one of its subcomponent has FAILED
- otherwise, If you see SKIPPED as the result for a component, at least one of its subcomponent has SKIPPED
- else, all components have been vested and passed the test.